include/ofi_coll.h

```diff
@@ -104,52 +108,68 @@ struct util_coll_reduce_item {
 	enum fi_op		op;
 };
 
+struct util_coll_mc {
+	struct fid_mc		mc_fid;
+	struct fid_ep		*ep;
+	struct util_av_set	*av_set;
+	struct slist		barrier_list;
+	struct slist		deferred_list;
+	struct slist		pending_xfer_list;
+	int			my_rank;
+	uint16_t		cid;
+	uint16_t		tag_seq;
+	ofi_atomic32_t		ref;
+};
+
 struct util_coll_comp_item;
 
 typedef void (*util_coll_comp_t)(struct util_coll_mc *coll_mc,
 				 struct util_coll_comp_item *comp);
 
-struct util_coll_join_comp_data {
+struct util_coll_comp_data {
 	uint64_t		cid_buf[OFI_CONTEXT_ID_SIZE];
 	uint64_t		tmp_cid_buf[OFI_CONTEXT_ID_SIZE];
 };
 
+struct util_coll_state {
+	struct dlist_entry	entry;
+	struct util_coll_mc	*coll_mc;
+	int			complete;
+};
```

Comment:
Does this need to be a separate structure, or can the fields just be added directly to util_coll_comp_item, or whatever structure is tracking the overall collective operation?